src/exceptions.jl

```diff
@@ -28,6 +93,295 @@ end
 
+function print_object_lost(io::IO, ctx::ObjectContext)
+    print(io, "Failed to retrieve object $(ctx.object_ref_hex). ")
+
+    # TODO: Support reporting call_site information
+    # if !isempty(ex.call_site)
+    #     print(io, "The `ObjectRef` was created at: $(ex.call_site)")
+    # else
+    #     print(io, "To see information about where this `ObjectRef` was created in Julia, " *
+    #               "set the environment variable RAY_record_ref_creation_sites=1 during " *
+    #               "`ray start` and `Ray.init()`.")
+    # end
+    # print(io, "\n\n")
+
+    return nothing
+end
+
+"""
+    ActorPlacementGroupRemoved <: RayError
+
+Raised when the corresponding placement group was removed.
+"""
+struct ActorPlacementGroupRemoved <: RayError end
+
+function Base.showerror(io::IO, ex::ActorPlacementGroupRemoved)
+    print(io, "$ActorPlacementGroupRemoved: ")
+    print(io, "The placement group corresponding to this Actor has been removed.")
+    return nothing
+end
+
+"""
+    ActorUnschedulableError <: RayError
+
+Raised when the actor cannot be scheduled.
+
+One example is that the node specified through NodeAffinitySchedulingStrategy is dead.
+"""
+struct ActorUnschedulableError <: RayError
+    msg::String
+end
+
+function Base.showerror(io::IO, ex::ActorUnschedulableError)
+    print(io, "$ActorUnschedulableError: ")
+    print(io, "The actor is not schedulable: $(ex.msg)")
+    return nothing
+end
+
+"""
+    LocalRayletDiedError <: RayError
+
+Indicates that the task's local raylet died.
+"""
+struct LocalRayletDiedError <: RayError end
+
+function Base.showerror(io::IO, ::LocalRayletDiedError)
+    print(io, "$LocalRayletDiedError: ")
+    print(io, "The task's local raylet died. Check raylet.out for more information.")
+    return nothing
+end
+
+"""
+    NodeDiedError <: RayError
+
+Indicates that the node is either dead or unreachable.
+"""
+struct NodeDiedError <: RayError
+    msg::String
+end
+
+function Base.showerror(io::IO, ex::NodeDiedError)
+    print(io, "$NodeDiedError: $(ex.msg)")
+    return nothing
+end
+
+"""
+    ObjectFetchTimedOutError <: RayError
+
+Indicates that an object fetch timed out.
+"""
+struct ObjectFetchTimedOutError <: RayError
+    object_context::ObjectContext
+end
+
+function Base.showerror(io::IO, ex::ObjectFetchTimedOutError)
+    print(io, "$ObjectFetchTimedOutError: ")
+    print_object_lost(io, ex.object_context)
+    print(io, m"""
+              Fetch for object $(ex.object_context.object_ref_hex) timed out because no
+              locations were found for the object. This may indicate a system-level bug.
+              """)
+
+    return nothing
+end
+
+"""
+    ObjectFreedError <: RayError
+
+Indicates that an object was manually freed by the application.
+
+Currently should never happen as Ray.jl doesn't currently implement `free` like
+[Ray for Python does](https://github.com/ray-project/ray/blob/ray-2.5.1/python/ray/_private/internal_api.py#L170).
```

Comment:
ah sonds like something we _should_ implement though (woudl help with some of the "disk full" errors we've seen during benchmarking big cluster jobs...)

Reply:
https://github.com/beacon-biosignals/Ray.jl/issues/230